src/Presentation/ListFacetHtmlBuilder.php

```diff
@@ -98,19 +98,33 @@ private function buildCheckboxesViewModel( FacetConfig $config, PropertyConstrai
 
 		$selectedValues = $combineWithAnd ? $state->getAndSelectedValues() : $state->getOrSelectedValues();
 
-		$checkboxes = [];
+		$visibleCheckboxes = [];
+		$collapsedCheckboxes = [];
 
 		foreach ( $this->getValuesAndCounts( $config ) as $i => $valueCount ) {
-			$checkboxes[] = [
+			$checkbox = [
 				'label' => $valueCount->value,
 				'count' => $valueCount->count,
 				'checked' => in_array( $valueCount->value, $selectedValues ), // TODO: test with multiple types of values
 				'value' => $valueCount->value,
 				'id' => $state->propertyId->getSerialization() . "-$i",
 			];
+
+			// TODO: Make the number of visible checkboxes configurable
+			if ( $i < 5 ) {
+				$visibleCheckboxes[] = $checkbox;
+			} else {
+				$collapsedCheckboxes[] = $checkbox;
+			}
 		}
 
-		return $checkboxes;
+		return [
+			'visible' => $visibleCheckboxes,
+			'collapsed' => $collapsedCheckboxes,
+			'showMore' => count( $collapsedCheckboxes ) > 0,
```

Comment:
Looks like you can return the method to its original form (returning just $checkboxes) and then have the caller (perhaps a new method) do

```php
return [
'visible' => array_slice_or_something_like_that(),
'collapsed' => array_slice_or_something_like_that(),
'showMore' => count( $checkboxes ) > 5,
];
```

Reply:
That separates concerns better

Also makes so existing tests can stay focused on building of the checkbox view model, without having to care about the new logic.